spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -439,7 +444,7 @@ public Cursor getCursor() {
 
 			structuredQueryBuilder.setKind(PartTreeDatastoreQuery.this.datastorePersistentEntity.kindName());
 
-			singularResult = (!isCountingQuery && collectionType == null) && !PartTreeDatastoreQuery.this.tree.isDelete();
+			singularResult = (!isCountingQuery && collectionType == null && !isStreamQuery) && !PartTreeDatastoreQuery.this.tree.isDelete();
```

Comment:
This boolean expression is getting too complicated. Since we are touching this line, could you refactor it into an if/else statement, with a view for reducing the negations ("if it's a counting query or a stream query, then singlarResult = false") is easier on the mind than "if it's not a counting query and it's not a stream query, singlarResult = true".

Reply:
Good point, I'll take the chance to make it easier to read.